synapse/http/federation/matrix_federation_agent.py

```diff
@@ -323,12 +323,17 @@ async def _resolve_server(self) -> List[Server]:
         if port or _is_ip_literal(host):
             return [Server(host, port or 8448)]
 
+        logger.debug("Looking up SRV record for %s", host)
```

Comment:
I believe that `host` is `bytes` here so this will end up logging:

> Looking up SRV record for b'foo'

I think you'll want to to `host.decode()` here to get it to a string.

This might also be true in other spots. 😄

Reply:
Right, I got tricked by that multiple times already.

Should be fixed.